tar.go

```diff
@@ -29,11 +30,13 @@ func NewTarStreamer(r io.Reader, keywords []string) Streamer {
 		tarReader:  tar.NewReader(pR),
 		keywords:   keywords,
 	}
-	go ts.readHeaders() // I don't like this
+
+	go ts.readHeaders()
```

Comment:
> By the time the archive is local, but not yet extracted, the manifest is available.

I don't see why your saying that. If you can do it in one pass as a *stream*, then creating a function that does it by reading things internally should work the same way -- no? And we could remove a bunch of the magic inside `tar.go`.